source/extensions/filters/http/ext_proc/processor_state.h

```diff
@@ -26,6 +37,8 @@ class ProcessorState : public Logger::Loggable<Logger::Id::filter> {
     HeadersCallback,
     // Waiting for a "body" response in buffered mode
     BufferedBodyCallback,
+    // Waiting for a streamed chunk
+    StreamedBodyCallback,
```

Comment:
For my education, why can't we get into the state where we're waiting for both headers and streamed chunk, e.g. at the start of a stream when we received headers/data independently as decoder callbacks?

Reply:
We can be in that state -- in that case, we are in "HeadersCallback" state and we buffer the data until the headers callback returns, then there is logic in ProcessorState::handleHeadersResponse to continue processing the buffered body depending on the body mode that's selected -- which might not have been known until the headers response came back.

Reply:
Ack; can you add some comments here explaining ^^ and this can be resolved..

Reply:
The comments are better now -- hope that helps.